src/base/QXmppJingleIq.cpp

```diff
@@ -120,6 +121,50 @@ static QString candidateToSdp(const QXmppJingleCandidate &candidate)
     return QStringLiteral("candidate:%1 %2 %3 %4 %5 %6 typ %7 generation %8").arg(candidate.foundation(), QString::number(candidate.component()), candidate.protocol(), QString::number(candidate.priority()), candidate.host().toString(), QString::number(candidate.port()), QXmppJingleCandidate::typeToString(candidate.type()), QString::number(candidate.generation()));
 }
 
+/// \cond
+///
+/// Parses the RTP Feedback Negotation elements.
+///
+/// \param parent element to parse which might contain RTP Feedback Negotiation elements
+/// \param properties RTP feedback properties to be set
+/// \param intervals RTP feedback intervals to be set
+///
```

Comment:
why are you writing doxygen documentation for code that is private and excluded from the generated doxygen documentation?

Reply:
It is the way we use for documenting code. I added `\cond` and `\endcond` in order to exclude it from the generated (public) documentation. How should / could that be done otherwise?

Reply:
Okay, so here comes my analysis about such documentation:

Why do you think this documentation is helpful here? More documentation can also bring disadvantages, it could get outdated and can mean more maintenance effort. Which new information does the documentation provide here?

When reading this documentation
 * you are already seeing the whole function implementation, because we don't generate external documentation
 * you already know the function name and the parameters

The current doxygen documentation provides:
* `/// Parses the RTP Feedback Negotation elements.`
  The name of the function, without 'Jingle', without camel-case and with spaces.
* `/// \param parent element to parse which might contain RTP Feedback Negotiation elements`
  That this elment is the parent is already given by the name of the parameter, the only new information here is that it *might contain such elements*
* `/// \param properties RTP feedback properties to be set`
  `/// \param intervals RTP feedback intervals to be set`
  New information: the function writes to these referenced variables, which is also indicated by `&` without `const` (but not explicit)

To sum up, the new information is:
 1. `parent` is allowed to contain no valid elements
 1. it's now clearer that the function writes to `properties` and `intervals`

Have I missed something?

Our goals are:
 * to minimize maintenance efforts, so not repeating information and making it easy to update the documentation when changing the code
 * to provide concise and easy-to-read information about the function, so only providing what is actually helpful and again not repeating information
 * and to minimize the development time which can also be important

The significant information we have got here can be put into a normal comment.
It could look like this:
```cpp
// Parses all found property and interval elements into properties and intervals.
static void parseJingleRtpFeedbackNegotiationElements(const QDomElement &parent, QVector<QXmppJingleRtpFeedbackProperty> &properties, QVector<QXmppJingleRtpFeedbackInterval> &intervals)
{
    for (auto child = parent.firstChildElement();
         !child.isNull();
         child = child.nextSiblingElement()) {
...
    }
}
```

The alternative would be to add the comments directly at the lines where the action is going on. This has the advantage that the comments are less likely forgot when changing the function. However in this case the comments seem completely useless:
```cpp
static void parseJingleRtpFeedbackNegotiationElements(const QDomElement &parent, QVector<QXmppJingleRtpFeedbackProperty> &properties, QVector<QXmppJingleRtpFeedbackInterval> &intervals)
{
    // parent may contain no valid child elements
    for (auto child = parent.firstChildElement();
         !child.isNull();
         child = child.nextSiblingElement()) {
        // Write parsed elements to properties and intervals
        if (QXmppJingleRtpFeedbackProperty::isJingleRtpFeedbackProperty(child)) {
            QXmppJingleRtpFeedbackProperty property;
            property.parse(child);
            properties.append(property);
        } else if (QXmppJingleRtpFeedbackInterval::isJingleRtpFeedbackInterval(child)) {
            QXmppJingleRtpFeedbackInterval interval;
            interval.parse(child);
            intervals.append(interval);
        }
    }
}
```

Now the thing is
 * this function is very simple
 * it's really not hard to see both points from the comment in the code
 * if the behaviour of the function changes, the comments need to be adjusted
 * if someone changes the behaviour and forgets to update the comments, the code will contain comments that tell something misleading
 
Did I missing something? Are there advantages of the doxygen styled documentation with explicit `\param`s?

In this case I don't think that the overhead of the full doxygen documentation stays in a good relation to the advantages. A simple comment does the same job and the important information is not repeated. I will leave the decision whether such a comment is needed here at all open to you.

Reply:
That's a point. Doesn't qtcreator display hints for normal comments and does it display doxygen inside \cond? I personally don't really use the hints.

For simple functions or functions in the same file I'm still not really convinced.